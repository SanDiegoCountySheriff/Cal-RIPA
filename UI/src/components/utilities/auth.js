import { PublicClientApplication } from '@azure/msal-browser'

export const doSignIn = async () => {
  await this.$msalInstance
    .loginPopup({})
    .then(() => {
      const myAccounts = this.$msalInstance.getAllAccounts()
      this.account = myAccounts[0]
      this.$emitter.emit('login', this.account)
    })
    .catch(error => {
      console.error(`error during authentication: ${error}`)
    })
}
