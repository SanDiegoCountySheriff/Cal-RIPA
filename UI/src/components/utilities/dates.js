export const formatDate = dateStr => {
  if (dateStr && dateStr.length > 0) {
    const str = dateStr.replace(/-/g, '')
    const year = str.substring(0, 4)
    const month = str.substring(4, 6)
    const day = str.substring(6, 8)
    return `${year}-${month}-${day}`
  }

  return ''
}
