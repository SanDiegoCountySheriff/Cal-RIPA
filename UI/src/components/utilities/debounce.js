export const debounce = (func, delay, immediate) => {
  let timerId
  return (...args) => {
    const boundFunc = func.bind(this, ...args)
    clearTimeout(timerId)
    if (immediate && !timerId) {
      boundFunc()
    }
    const calleeFunc = immediate
      ? () => {
          timerId = null
        }
      : boundFunc
    timerId = setTimeout(calleeFunc, delay)
  }
}
