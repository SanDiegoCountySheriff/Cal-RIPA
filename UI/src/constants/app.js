export const VERSION = '#{TEMPLATE_VERSION}#'
