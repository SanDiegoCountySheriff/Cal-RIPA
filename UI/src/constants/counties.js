export const COUNTIES = [
  {
    name: 'Alameda',
  },
  {
    name: 'Alpine',
  },
  {
    name: 'Amador',
  },
  {
    name: 'Butte',
  },
  {
    name: 'Calaveras',
  },
  {
    name: 'Colusa',
  },
  {
    name: 'Contra Costa',
  },
  {
    name: 'Del Norte',
  },
  {
    name: 'El Dorado',
  },
  {
    name: 'Fresno',
  },
  {
    name: 'Glenn',
  },
  {
    name: 'Humboldt',
  },
  {
    name: 'Imperial',
  },
  {
    name: 'Inyo',
  },
  {
    name: 'Kern',
  },
  {
    name: 'Kings',
  },
  {
    name: 'Lake',
  },
  {
    name: 'Lassen',
  },
  {
    name: 'Los Angeles',
  },
  {
    name: 'Madera',
  },
  {
    name: 'Marin',
  },
  {
    name: 'Mariposa',
  },
  {
    name: 'Mendocino',
  },
  {
    name: 'Merced',
  },
  {
    name: 'Modoc',
  },
  {
    name: 'Mono',
  },
  {
    name: 'Monterey',
  },
  {
    name: 'Napa',
  },
  {
    name: 'Nevada',
  },
  {
    name: 'Orange',
  },
  {
    name: 'Placer',
  },
  {
    name: 'Plumas',
  },
  {
    name: 'Riverside',
  },
  {
    name: 'Sacramento',
  },
  {
    name: 'San Benito',
  },
  {
    name: 'San Bernardino',
  },
  {
    name: 'San Diego',
  },
  {
    name: 'San Francisco',
  },
  {
    name: 'San Joaquin',
  },
  {
    name: 'San Luis Obispo',
  },
  {
    name: 'San Mateo',
  },
  {
    name: 'Santa Barbara',
  },
  {
    name: 'Santa Clara',
  },
  {
    name: 'Santa Cruz',
  },
  {
    name: 'Shasta',
  },
  {
    name: 'Sierra',
  },
  {
    name: 'Siskiyou',
  },
  {
    name: 'Solano',
  },
  {
    name: 'Sonoma',
  },
  {
    name: 'Stanislaus',
  },
  {
    name: 'Sutter',
  },
  {
    name: 'Tehama',
  },
  {
    name: 'Trinity',
  },
  {
    name: 'Tulare',
  },
  {
    name: 'Tuolumne',
  },
  {
    name: 'Ventura',
  },
  {
    name: 'Yolo',
  },
  {
    name: 'Yuba',
  },
]
