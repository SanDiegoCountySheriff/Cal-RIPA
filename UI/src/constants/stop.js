export const SUBMISSION_STATUSES = [
  {
    text: 'Failed',
    value: 'Failed',
    editable: true,
  },
  {
    text: 'Submitted',
    value: 'Submitted',
    editable: false,
  },
  {
    text: 'Unsubmitted',
    value: 'Unsubmitted',
    editable: true,
  },
  {
    text: 'Resubmitted',
    value: 'Resubmitted',
    editable: false,
  },
]
