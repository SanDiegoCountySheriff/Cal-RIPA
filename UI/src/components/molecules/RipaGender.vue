<template>
  <div class="ripa-gender tw-pb-8">
    <template class="tw-mb-4">
      <ripa-form-header
        title="Perceived Gender"
        required
        subtitle="ยง999.226(a)(5)"
      >
      </ripa-form-header>

      <ripa-radio-group
        v-model="model.perceivedGender"
        :items="genderItems"
        @input="handleInput"
      >
      </ripa-radio-group>
    </template>

    <template>
      <ripa-form-header
        class="tw-mt-8"
        title="Perceived LGBT"
        required
        subtitle="ยง999.226(a)(6)"
      >
      </ripa-form-header>

      <ripa-switch
        v-model="model.perceivedLgbt"
        label="Perceived as LGBT"
        :max-width="200"
        @input="handleInput"
      ></ripa-switch>
    </template>
  </div>
</template>

<script>
import RipaFormHeader from '@/components/molecules/RipaFormHeader'
import RipaRadioGroup from '@/components/atoms/RipaRadioGroup'
import RipaSwitch from '@/components/atoms/RipaSwitch'
import { GENDERS } from '@/constants/form'

export default {
  name: 'ripa-gender',

  components: {
    RipaFormHeader,
    RipaRadioGroup,
    RipaSwitch,
  },

  data() {
    return {
      valid: true,
      genderItems: GENDERS,
      viewModel: {
        perceivedGender: this.value?.perceivedGender || null,
        perceivedLgbt: this.value?.perceivedLgbt || false,
      },
    }
  },

  computed: {
    model: {
      get() {
        return this.viewModel
      },
    },
  },

  methods: {
    handleInput() {
      this.updatePerceivedLgbtModel()
      this.$emit('input', this.viewModel)
    },

    updatePerceivedLgbtModel() {
      if (
        this.viewModel.perceivedGender === 3 ||
        this.viewModel.perceivedGender === 4
      ) {
        this.viewModel.perceivedLgbt = true
      }
    },
  },

  props: {
    value: {
      type: Object,
      default: () => {},
    },
  },
}
</script>
