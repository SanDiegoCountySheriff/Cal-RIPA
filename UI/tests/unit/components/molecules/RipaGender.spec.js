import RipaGender from '@/components/molecules/RipaGender.vue'
import { shallowMount, mount } from '@vue/test-utils'
import { defaultStop } from '@/utilities/stop'
import Vuetify from 'vuetify'
import { V2_STOP } from '../../constants/RipaFormContainerTestConstants'

describe('Ripa Gender', () => {
  let vuetify
  let wrapper
  let stop

  beforeEach(() => {
    vuetify = new Vuetify()
    stop = defaultStop()
  })

  const factory = propsData => {
    return shallowMount(RipaGender, {
      vuetify,
      propsData: {
        ...propsData,
      },
    })
  }

  it('should match snapshot', () => {
    wrapper = mount(RipaGender, {
      vuetify,
      propsData: {
        value: stop,
      },
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should display nonBinaryPerson for v2 stop', () => {
    const v2Stop = V2_STOP
    v2Stop.person.nonBinaryPerson = true
    wrapper = mount(RipaGender, {
      vuetify,

      propsData: { value: v2Stop },

      provide: {
        isOnlineAndAuthenticated() {
          return true
        },

        lastResult() {
          return {}
        },
      },
    })

    expect(wrapper.html()).toContain('Nonbinary person')
  })

  it.skip('should display gender nonconforming for legacy stop', () => {
    wrapper = mount(RipaGender, {
      vuetify,

      propsData: { value: stop },

      provide: {
        isOnlineAndAuthenticated() {
          return true
        },

        lastResult() {
          return {}
        },
      },
    })

    expect(wrapper.html()).toContain('Gender Nonconforming')
  })
})
