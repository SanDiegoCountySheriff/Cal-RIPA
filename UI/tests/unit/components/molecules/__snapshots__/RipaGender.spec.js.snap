// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Ripa Gender should match snapshot 1`] = `
<div class="ripa-gender tw-pb-4">
  <div class="ripa-form-header">
    <div class="row tw-ml-1 tw-mt-1"><span class="v-tooltip v-tooltip--top"><!----><i aria-hidden="true" aria-haspopup="true" aria-expanded="false" class="v-icon notranslate mdi mdi-alert theme--light error--text" style="font-size: 16px;"></i></span> <button type="button" class="v-icon notranslate icon v-icon--link mdi mdi-scale-balance theme--light primary--text" style="font-size: 16px;"></button>
      <div class="ripa-header"><span class="tw-text-base md:tw-text-lg tw-uppercase">Perceived Gender</span></div>
    </div>
  </div>
  <div class="container">
    <div class="row no-gutters">
      <div class="col-sm-12 col-12">
        <div class="ripa-radio-group">
          <div class="v-input theme--light v-input--selection-controls v-input--radio-group v-input--radio-group--column">
            <div class="v-input__control">
              <div class="v-input__slot" style="height: auto;">
                <div role="radiogroup" aria-labelledby="input-8" class="v-input--radio-group__input">
                  <div class="v-radio theme--light">
                    <div class="v-input--selection-controls__input"><i aria-hidden="true" class="v-icon notranslate mdi mdi-radiobox-blank theme--light"></i><input aria-checked="false" id="input-9" role="radio" type="radio" name="radio-8" value="1">
                      <div class="v-input--selection-controls__ripple"></div>
                    </div><label for="input-9" class="v-label theme--light" style="left: 0px; position: relative;">Male</label>
                  </div>
                  <div class="v-radio theme--light">
                    <div class="v-input--selection-controls__input"><i aria-hidden="true" class="v-icon notranslate mdi mdi-radiobox-blank theme--light"></i><input aria-checked="false" id="input-11" role="radio" type="radio" name="radio-8" value="2">
                      <div class="v-input--selection-controls__ripple"></div>
                    </div><label for="input-11" class="v-label theme--light" style="left: 0px; position: relative;">Female</label>
                  </div>
                  <div class="v-radio theme--light">
                    <div class="v-input--selection-controls__input"><i aria-hidden="true" class="v-icon notranslate mdi mdi-radiobox-blank theme--light"></i><input aria-checked="false" id="input-13" role="radio" type="radio" name="radio-8" value="3">
                      <div class="v-input--selection-controls__ripple"></div>
                    </div><label for="input-13" class="v-label theme--light" style="left: 0px; position: relative;">Transgender Male</label>
                  </div>
                  <div class="v-radio theme--light">
                    <div class="v-input--selection-controls__input"><i aria-hidden="true" class="v-icon notranslate mdi mdi-radiobox-blank theme--light"></i><input aria-checked="false" id="input-15" role="radio" type="radio" name="radio-8" value="4">
                      <div class="v-input--selection-controls__ripple"></div>
                    </div><label for="input-15" class="v-label theme--light" style="left: 0px; position: relative;">Transgender Female</label>
                  </div>
                </div>
              </div>
              <div class="v-messages theme--light">
                <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
              </div>
            </div>
          </div>
          <!---->
        </div>
      </div>
      <div class="col-sm-12 col-12">
        <div class="ripa-switch" style="max-width: 300px;">
          <div class="v-input v-input--reverse v-input--expand theme--light v-input--selection-controls v-input--switch v-input--switch--inset">
            <div class="v-input__control">
              <div class="v-input__slot">
                <div class="v-input--selection-controls__input"><input aria-checked="false" id="input-20" role="switch" type="checkbox" aria-disabled="false" value="">
                  <div class="v-input--selection-controls__ripple"></div>
                  <div class="v-input--switch__track theme--light"></div>
                  <div class="v-input--switch__thumb theme--light">
                    <transition-stub name="fab-transition" mode="out-in"></transition-stub>
                  </div>
                </div><label for="input-20" class="v-label theme--light" style="left: 0px; position: relative;"><span class="main-label">
        Gender Nonconforming
      </span> <span class="switch-label" style="right: -45px;">
        No
      </span></label>
              </div>
              <div class="v-messages theme--light">
                <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
              </div>
            </div>
          </div>
        </div>
        <!---->
      </div>
    </div>
  </div>
  <div class="ripa-form-header tw-mt-8">
    <div class="row tw-ml-1 tw-mt-1"><span class="v-tooltip v-tooltip--top"><!----><i aria-hidden="true" aria-haspopup="true" aria-expanded="false" class="v-icon notranslate mdi mdi-alert theme--light error--text" style="font-size: 16px;"></i></span> <button type="button" class="v-icon notranslate icon v-icon--link mdi mdi-scale-balance theme--light primary--text" style="font-size: 16px;"></button>
      <div class="ripa-header"><span class="tw-text-base md:tw-text-lg tw-uppercase">Perceived LGBT</span></div>
    </div>
  </div>
  <div class="container">
    <div class="row no-gutters">
      <div class="col-sm-12 col-12">
        <div class="ripa-switch" style="max-width: 300px;">
          <div class="v-input v-input--reverse v-input--expand v-input--hide-details theme--light v-input--selection-controls v-input--switch v-input--switch--inset">
            <div class="v-input__control">
              <div class="v-input__slot">
                <div class="v-input--selection-controls__input"><input aria-checked="false" id="input-30" role="switch" type="checkbox" aria-disabled="false" value="">
                  <div class="v-input--selection-controls__ripple"></div>
                  <div class="v-input--switch__track theme--light"></div>
                  <div class="v-input--switch__thumb theme--light">
                    <transition-stub name="fab-transition" mode="out-in"></transition-stub>
                  </div>
                </div><label for="input-30" class="v-label theme--light" style="left: 0px; position: relative;"><span class="main-label">
        Perceived as LGBT
      </span> <span class="switch-label" style="right: -45px;">
        No
      </span></label>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
